src/AbortablePromiseCache.ts

```diff
@@ -198,7 +198,7 @@ export default class AbortablePromiseCache<T, U> {
     // if we got here, it is not in the cache. fill.
     this.fill(key, data, signal, statusCallback)
     return AbortablePromiseCache.checkSinglePromise(
-      this.cache.get(key).promise,
+      this.cache.get(key)!.promise,
```

Comment:
what does the exclamation mark (!.) do here?

Reply:
It's a [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-). We know that `this.cache.get(key)` won't return `undefined` since we just filled the cache for that key a couple lines above, but TypeScript doesn't know that, so we have to convince it with `!`.